### ISSUE TITLE
'Expand window' button missing from 'Shuffle Workflow' AND 'User Input' execution argument field

### ISSUE THREAD

**Issue by deb-alex** [Shuffle/Shuffle#1074]
Most text fields across Shuffle have an 'Expand window' button, which enables the user to input text conveniently in a large window with a data previewer.

![image](https://user-images.githubusercontent.com/128032364/228246974-ebca55c6-2ed4-4d0c-aeb7-828d9824b6df.png)

This button is missing from the 'Shuffle Workflow' execution argument field.
![image](https://user-images.githubusercontent.com/128032364/228246715-3f835be6-34a7-4d7f-84d2-4168aef26661.png)

This makes it impractical to manually input large amounts of data for the execution argument.

**Comment by frikky** [comment]
> Most text fields across Shuffle have an 'Expand window' button, which enables the user to input text conveniently in a large window with a data previewer.
> 
> ![image](https://user-images.githubusercontent.com/128032364/228246974-ebca55c6-2ed4-4d0c-aeb7-828d9824b6df.png)
> 
> This button is missing from the 'Shuffle Workflow' execution argument field. ![image](https://user-images.githubusercontent.com/128032364/228246715-3f835be6-34a7-4d7f-84d2-4168aef26661.png)
> 
> This makes it impractical to manually input large amounts of data for the execution argument.

Yep, you're right! We're reframing how the "Expand Window" button is being called as we plan to make it available for more fields than just these (e.g. variables, user input, schedule, webhook etc..)

Adding as part of 1.3.0 as we won't have time before 1.2.0, where we're focusing on stability & scale.

**Comment by rhithesh** [comment]
can you specify which specific app has the execution argument most of them have call and api. @0x0elliot  
I tried to go through a few apps but they did not have the execution argument field a screenshot would be helpful. thanks.

**Comment by 0x0elliot** [comment]
@rhithesh the shuffle-tools app would if you select execute python

**Comment by frikky** [comment]
> can you specify which specific app has the execution argument most of them have call and api. @0x0elliot I tried to go through a few apps but they did not have the execution argument field a screenshot would be helpful. thanks.

@rhithesh 

ALL apps have the button that is "expand window" next to all their fields: 
<img width="674" alt="image" src="https://github.com/Shuffle/Shuffle/assets/5719530/62b9ff8e-eccd-46ba-9815-ff395d292c17">

This does not go for Triggers however (find them in their own tab at the bottom left of any workflow):
<img width="674" alt="image" src="https://github.com/Shuffle/Shuffle/assets/5719530/18720b2c-3931-46e2-8163-bb4507477b24">

The point is to add the same popout window to these as is for apps to make it easy to edit complex things in them.

**Comment by rhithesh** [comment]
hi @frikky 

<img width="1431" alt="Screenshot 2023-11-19 at 11 27 40 PM" src="https://github.com/Shuffle/Shuffle/assets/115660976/01adf37d-06aa-4746-9801-dc964e0d9ab1">

you can see the expand icon here.

<img width="1431" alt="Screenshot 2023-11-19 at 11 27 51 PM" src="https://github.com/Shuffle/Shuffle/assets/115660976/d9ad3c92-9e16-40a1-bd75-b51582264efc">

after the onClick 


<img width="1431" alt="Screenshot 2023-11-19 at 11 28 16 PM" src="https://github.com/Shuffle/Shuffle/assets/115660976/192ccf28-d6cf-4dd7-aef8-cad9a648a63b">

error when trying to submit and also I am not able to do data persistence 




AngularWorkflow.jsx ,  ShuffelCodeEditor.jsx are the two components I have made changes to I am not able to understand the role of the props of  ShuffelCodeEditor  such actionlist setEditordata values are  is it okay if I can do a draft pr, can you also give me some insight on how ShuffelCodeEditor props play a role.  thanks

**Comment by frikky** [comment]
Looks like you're on the right track, but the reason this hasn't been implemented yet is exactly this: it's trickier than it seems.  State management in this UI is a mess, and expanding functionality always leads to things like this.

Become friendly with the in-browser debugger (or console log ALOT) and you can solve this for sure!

**Comment by rhithesh** [comment]
@frikky I have added comments please look into them thanks.

**Comment by frikky** [comment]
> @frikky I have added comments please look into them thanks.

Hey! How far did we get with this? Seeing as the draft is still open :)

**Comment by satti-hari-krishna-reddy** [comment]
Hey @rhithesh , how's it going with the issue you're working on? Any progress or challenges you're facing?

**Comment by frikky** [comment]
Fixed!
